streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -251,26 +252,69 @@ public StreamTask(final TaskId id,
         if (eosEnabled) {
             initializeTransactions();
         }
-    }
 
-    @Override
-    public boolean initializeStateStores() {
-        log.debug("Initializing state stores");
+        try {
+            final Map<TopicPartition, OffsetAndMetadata> offsetsAndMetadata = consumer.committed(partitions);
+            initializeCommittedOffsets(offsetsAndMetadata);
+            initializeTaskTime(offsetsAndMetadata);
```

Comment:
Apparently it is not completely save to call instance methods from the constructor since the object and its member variables are  not completely initialised yet. Static methods would be fine, though.

I do not think that `initializeCommittedOffsets()` is currently problematic since it only accesses member variables from `AbstractTask` which should be already initialised. However, `initializeTaskTime()` accesses `partitionGroup` which is created in the constructor. Since we cannot know what reorderings a compiler does, it would be better to specify `initializeTaskTime()` as static and pass in `partitionGroup`. To make `initializeCommittedOffsets()` future-proof, it would also be good to also transform it to a static method.